Source/Core/InputCommon/ControllerInterface/ControllerInterface.cpp

```diff
@@ -299,7 +299,8 @@ void ControllerInterface::ClearDevices()
     m_devices.clear();
   }
 
-  InvokeDevicesChangedCallbacks();
+  if (!m_populating_devices_counter)
```

Comment:
Perhaps:

```suggestion
  if (m_populating_devices_counter.load(std::memory_order_relaxed) == 0)
```

Since this is just querying the value of it